static/app/views/settings/account/apiApplications/index.tsx

```diff
@@ -7,90 +7,99 @@ import {
 } from 'sentry/actionCreators/indicator';
 import {Button} from 'sentry/components/button';
 import EmptyMessage from 'sentry/components/emptyMessage';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
 import Panel from 'sentry/components/panels/panel';
 import PanelBody from 'sentry/components/panels/panelBody';
 import PanelHeader from 'sentry/components/panels/panelHeader';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import {IconAdd} from 'sentry/icons';
 import {t} from 'sentry/locale';
 import type {ApiApplication} from 'sentry/types';
-import DeprecatedAsyncView from 'sentry/views/deprecatedAsyncView';
+import {setApiQueryData, useApiQuery, useQueryClient} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
 import Row from 'sentry/views/settings/account/apiApplications/row';
 import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
 
 const ROUTE_PREFIX = '/settings/account/api/';
 
 type Props = RouteComponentProps<{}, {}>;
-type State = {
-  appList: ApiApplication[];
-} & DeprecatedAsyncView['state'];
 
-class ApiApplications extends DeprecatedAsyncView<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncView['getEndpoints']> {
-    return [['appList', '/api-applications/']];
-  }
-
-  getTitle() {
-    return t('API Applications');
-  }
+function ApiApplications({router}: Props) {
+  const api = useApi();
+  const queryClient = useQueryClient();
 
-  handleCreateApplication = async () => {
+  const handleCreateApplication = async () => {
     addLoadingMessage();
 
     try {
-      const app = await this.api.requestPromise('/api-applications/', {
+      const app = await api.requestPromise('/api-applications/', {
         method: 'POST',
       });
 
       addSuccessMessage(t('Created a new API Application'));
-      this.props.router.push(`${ROUTE_PREFIX}applications/${app.id}/`);
+      router.push(`${ROUTE_PREFIX}applications/${app.id}/`);
     } catch {
       addErrorMessage(t('Unable to remove application. Please try again.'));
     }
   };
 
-  handleRemoveApplication = (app: ApiApplication) => {
-    this.setState({
-      appList: this.state.appList.filter(a => a.id !== app.id),
-    });
+  const handleRemoveApplication = (app: ApiApplication) => {
+    setApiQueryData<any>(queryClient, ['/api-applications/'], oldAppList =>
```

Comment:
maybe pull `['/api-applications/']` into a constant and share it between the request and setApiQueryData